### ISSUE TITLE
DPL-500 View and edit retention instructions on a labware

### ISSUE THREAD

**Issue by KatyTaylor** [sanger/sequencescape#3676]
**User story**

As a Sample Management user, I want to be able to edit the 'Retention instructions' on a labware, so that I can manually enter this information against the stock plates made from extraction tubes (we only get 2 or 3 of these every couple of months, so the manual work is not too much).

**Who are the primary contacts for this story**
Danni W, Siobhan A-G, Katy

**Who is the nominated tester for UAT**
Danni W, Siobhan A-G

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] I can update the 'Retention instructions' for a particular labware
- [x] I can click "Edit Retention Instructions" link on the right hand side, go to a new page to edit retention instruction and save the update.
- [x] I can add a retention instruction for labware that currently does not have a retention instruction.
- [x] Users other than **admin** privileges **cannot** edit retention instructions.

**Dependencies**
This story is blocked by the following dependencies:
- https://github.com/sanger/sequencescape/issues/3673
- https://github.com/sanger/sequencescape/issues/4093

References
This story has a non-blocking relationship with:
- https://github.com/sanger/sequencescape/issues/3675
- https://github.com/sanger/sequencescape/issues/4094
- https://github.com/sanger/sequencescape/issues/4095

**Additional Context**

Note that this story focuses specifically on the user-interface. Check the **References** section for the related stories.

**Comment by SujitDey2022** [comment]
Need to review the occurrence of a request to update the value. Also, need to validate the correct values to be shown to the user and identify if the change can be made in bulk v/s per plate. As the policy is new, the assumption is that the SSRs may have to manually re-upload the manifest for each plate/tube.

**Comment by SujitDey2022** [comment]
Refinement 21/11/23:

1. Limit Edit access to Admin roles in SS
2. May need a separate story for identifying how the retention data is stored
3. Record who and when made the change and is viewable in the plate event section
4. Changes to apply at the stock plate level
5. Changes to be verified for samples in tubes as well

**Comment by KatyTaylor** [comment]
This was implemented in https://github.com/sanger/sequencescape/issues/3673 using custom metadata.

In order to satisfy the requirement "Limit Edit access to Admin roles in SS", we might need to change where the data is stored - because custom metadata is a key-value pair store, and we wouldn't want to limit edit access to ALL custom metadata, just retention instructions.

**Comment by SujitDey2022** [comment]
09Jan2024 - Team Discussion

1. Research on how and where metadata is stored from point of view admin access to editing. (New story)
2. Limit Edit access to Admin roles should be across SS and Limber

**Comment by KatyTaylor** [comment]
@dasunpubudumal think this could do with updating, following our recent decisions. 
- remove the italic text from the top
- actually you've added the 'view' functionality to https://github.com/sanger/sequencescape/issues/4093 already - so this could just be about the 'edit' functionality
- add the UX idea for it to be a separate page accessed via a link on the RHS of the page?

**Comment by dasunpubudumal** [comment]
@KatyTaylor I updated the description :)

**Comment by dasunpubudumal** [comment]
**A note about existing labware that has retention instructions stored in `custom_metadata`**

- **In perspective of the Asset Summary**: When displaying the asset summary (see the screenshot below), the sequence of procedures is as follows:
    1. Check if `labware.retention_instruction` is populated. If it is, display its value.
    2. If not, check if there's a corresponding retention instruction in `custom_metadata`. If there is, it will display its value.
    3. Else, the value will be displayed as "**Not Set**".
<div align="center">
<img width="602" alt="image" src="https://github.com/sanger/sequencescape/assets/25484917/ce04ab86-af93-4cf8-8297-0c0917e5fa7c">
</div>

-  **In perspective of the "Edit Retention Instruction" page**: When displaying the update page (see the screenshot below), the user is able to select any retention instruction, as in the asset summary. The initial value in the dropdown (i.e., `select`) is set to be the value in `labware.retention_instruction`. If `labware.retention_instruction` is not populated, it will check for the `custom_metadata` table and set the corresponding retention instruction.
<div align="center">
<img width="362" alt="image" src="https://github.com/sanger/sequencescape/assets/25484917/3a374ed3-eee9-44c1-be33-c0d3ef8df06c">
</div>
